gateway-abac/templates/statefulset.yaml

```diff
@@ -69,32 +69,34 @@ spec:
           - configMapRef:
               name: {{ include "gateway.fullname" . }}
           env:
-          {{- if .Values.saslDownstreamCredentials }}
+          {{- if or .Values.saslDownstreamCredentials (and .Values.existingSecret.name .Values.existingSecret.saslDownstreamCredentialsName) }}
           - name: GATEWAY_SMTP_SASL_ACCOUNTS
             valueFrom:
               secretKeyRef:
-                key: gateway-sasl-auth-downstream
-                name: "{{ include "gateway.fullname" . }}"
+                key: {{ include "gateway.saslDownstreamCredentialsKey" . }}
+                name: {{ include "gateway.secretName" . }}
           {{- end }}
-          {{- if .Values.saslUpstreamCredentials }}
+          {{- if or .Values.saslUpstreamCredentials (and .Values.existingSecret.name .Values.existingSecret.saslUpstreamCredentialsName) }}
           - name: GATEWAY_SMTPD_SASL_ACCOUNTS
             valueFrom:
               secretKeyRef:
-                key: gateway-sasl-auth-upstream
-                name: "{{ include "gateway.fullname" . }}"
+                key: {{ include "gateway.saslUpstreamCredentialsKey" . }}
+                name: {{ include "gateway.secretName" . }}
           {{- end }}
-          {{- if .Values.xheaderAuthSecret }}
+          {{- if or .Values.xHeaderAuthSecret (and .Values.existingSecret.name .Values.existingSecret.xHeaderAuthSecretName) }}
```

Comment:
Changing the the existing value key from `xheaderAuthSecret` to `xHeaderAuthSecret` makes this a breaking change, meaning the chart should bump to 2.x. Since it's just a minor change, any objection to just leaving this as `xheaderAuthSecret`?